homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -328,21 +344,20 @@ def _async_button_event(button_id, event_type):
             # use the parent_device for HA device info
             ha_device = bridge_devices[device["parent_device"]]
 
-        ha_device_serial = _handle_none_keypad_serial(
-            ha_device, bridge_devices[BRIDGE_DEVICE_ID]["serial"]
-        )
+        device_info = device_info_by_device_id.get(ha_device["device_id"])
 
         type_ = _lutron_model_to_device_type(ha_device["model"], ha_device["type"])
         area = _area_name_from_id(bridge_device.areas, ha_device["area"])
-        name = ha_device["name"].split("_")[-1]
+        name = device_info["name"]
+        serial = next(iter(device_info["identifiers"]))[1]
```

Comment:
Not sure what you'll think of this line.
Best result I could find by googling how to pull the first(only) item out of a set with minimal overhead.
I'm open to a better suggestion

Reply:
I know I don't want to use .pop() as it would alter the set

Reply:
Actual serial numbers are in the dict, but not the ones we're forced to generate for non-serialized keypads.  
I figured it would be best to only generate it once, and then just retrieve it from the DeviceInfo of the Keypad.  
Also goes for the keypad name too, instead of regenerating it, just retrieve it.